numba_rvsdg/core/datastructures/basic_block.py

```diff
@@ -146,15 +169,23 @@ class SyntheticExitBranch(SyntheticBranch):
 @dataclass(frozen=True)
 class RegionBlock(BasicBlock):
     kind: str = None
-    headers: Dict[str, BasicBlock] = None
-    """The header of the region"""
+    """The kind of region. Can be 'head', 'tail', 'branch', 
+    'loop' or 'meta' strings"""
+    parent_region: "RegionBlock" = None
+    """The parent region of this region as per the SCFG."""
+    header: str = None
+    """The header node of the region"""
     subregion: "SCFG" = None
-    """The subgraph excluding the headers
-    """
+    """The subgraph as an independent SCFG. Note that in case
+    of subregions the exiting node may point to blocks outside
+    of the current SCFG object."""
     exiting: str = None
-    """The exiting node.
-    """
+    """The exiting node of the region."""
+
+    def replace_header(self, new_header):
+        """Does inplace replacement of the header block."""
+        object.__setattr__(self, "header", new_header)
```

Comment:
That's a hack to bypass frozen dataclass's immutability

Reply:
This relates to an issue that arises when treating `Regions` as blocks. i.e. When we 'wrap' a block in a `RegionBlock` and if that block is the header of some other region, technically the header of that parent region is being changed.